app/questionnaire/router.py

```diff
@@ -270,16 +271,30 @@ def _is_section_enabled(self, section):
         if "enabled" not in section:
             return True
 
-        for condition in section["enabled"]:
-            if evaluate_when_rules(
+        enabled = section["enabled"]
+        if isinstance(enabled, dict):
+            when_rule_evaluator = WhenRuleEvaluator(
+                self._schema,
+                self._answer_store,
+                self._list_store,
+                self._metadata,
+                self._response_metadata,
+                location=None,
+                routing_path_block_ids=None,
+            )
+
+            return when_rule_evaluator.evaluate(enabled["when"])
+
+        return any(
+            evaluate_when_rules(
                 condition["when"],
                 self._schema,
                 self._metadata,
                 self._answer_store,
                 self._list_store,
-            ):
-                return True
-        return False
+            )
+            for condition in enabled
+        )
```

Comment:
Nice use of `any`.

Reply:
I was not sure if you'll prefer simplicity over readability here but "simple is better than complex".